client/my-sites/customer-home/cards/features/help-search/index.jsx

```diff
@@ -52,7 +60,22 @@ const HelpSearch = ( { searchQuery, hideInlineHelpUI, showInlineHelpUI, openDial
 		// Grab properties using constants for safety
 		const resultPostId = get( result, RESULT_POST_ID );
 		const resultLink = amendYouTubeLink( get( result, RESULT_LINK ) );
+		const type = get( result, RESULT_TYPE, RESULT_ARTICLE );
+		const tour = get( result, RESULT_TOUR );
+
+		// Create and send tracking event
+		const tracksData = omitBy(
+			{
+				search_query: searchQuery,
+				tour,
+				result_url: resultLink,
+				location: HELP_COMPONENT_LOCATION,
+			},
+			isUndefined
+		);
+		track( `calypso_inlinehelp_${ type }_open`, tracksData );
```

Comment:
The method is named `openResultView` but only `resultPostId` and `resultLink` are actually required to call `openDialog` which is the key action of this method.

I wonder whether this entire block of tracking code could/should be moved to its own dedicated method?

Reply:
Sure, extracted `trackResultView` here: c3fd0b3e2.  Also extracted `getResultLink` since it seemed like too much magic to have in two places (`trackResultView` and `openResultView`).  What do you think?